Adding a test for within subjects designs via an aovlist object

ibecav opened the pull request:
-- failed to be able to use an object type of anova generating reprex
This successfully covers aov, lm and aovlist

IndrajeetPatil commented:
Hmm, strange that two of the tests are failing on my PC:

![image](https://user-images.githubusercontent.com/11330453/46495855-843f1e00-c7e4-11e8-8c6d-9c8723cb2b7e.png)

I have already merged the PR. Can you update your fork and see if you can reproduce this?

Maybe we are using different versions of `sjstats`? (I am using `0.17.1`, the recent version on `CRAN`).

IndrajeetPatil commented:
Yeah, this is not just on my local computer, the build is also failing on CI platforms:
https://ci.appveyor.com/project/IndrajeetPatil/ggstatsplot

ibecav commented:
I'm running 0.17.1.

I deleted my development branch and pulled your new master.

The problem is when you merged and cleaned up you deleted a bunch of `set.seed` that are actually important if you put them back in the tests will start passing again.



```
# model-1
set.seed(123)
df1 <- ggstatsplot:::lm_effsize_ci(
  object = lmobject1,
  partial = FALSE,
  conf.level = .99,
  nboot = 100
)

# model-2
set.seed(123)
df2 <- ggstatsplot:::lm_effsize_ci(
  object = lmobject2,
  partial = FALSE,
  conf.level = .99,
  nboot = 100
)

# model-3
set.seed(123)
df3 <- ggstatsplot:::lm_effsize_ci(
  object = lmobject3,
  partial = FALSE,
  conf.level = .99,
  nboot = 100
)

```

ibecav commented:
I guess I should have warned you in advance because I discovered that little oddity in my own development but I didn't notice you removed them...  It only impacts the confidence intervals of course because they rely on bootstrapping. Which is why I testing them.  It's hard to get things like the degrees of freedom wrong

IndrajeetPatil commented:
Huh, fascinating. Why does the seed need to be placed before every model? 
My understanding was that, once set with `set.seed()`, this seed value will be used for the rest of the active R session until it is modified further with another `set.seed()` call. But I am guessing I am misinformed about this?

ibecav commented:
Could not find a canonical answer on SO.  Closest I got was this.
https://stackoverflow.com/questions/18696331/how-many-times-to-use-set-seed123

All I know is that is definitely true and if you don't stick them in you
get very slight variations and if you do the problem goes away

On Thu, Oct 4, 2018 at 3:56 PM Indrajeet Patil <notifications@github.com>
wrote:

> Huh, fascinating. Why does the seed need to be placed before every model?
> My understanding was that, once set with set.seed(), this seed value will
> be used for the rest of the active R session until it is modified further
> with another set.seed() call. But I am guessing I am misinformed about
> this?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/IndrajeetPatil/ggstatsplot/pull/70#issuecomment-427148875>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AZU62S4yE8NanHimq6g69-cs28SfqKTpks5uhmflgaJpZM4XIlAg>
> .
>
